lib/workers/global/config/parse/file.ts

```diff
@@ -1,8 +1,25 @@
+import { readFileSync } from 'fs-extra';
+import { load } from 'js-yaml';
+import JSON5 from 'json5';
 import upath from 'upath';
 import { migrateConfig } from '../../../../config/migration';
-import type { AllConfig } from '../../../../config/types';
+import type { AllConfig, RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 
+export function getParsedContent(file: string): RenovateConfig {
+  const rawContent = readFileSync(file, 'utf8');
+  switch (upath.extname(file)) {
+    case '.yaml':
+    case '.yml':
+      return load(rawContent, { json: true }) as RenovateConfig;
+    case '.json5':
+      return JSON5.parse(rawContent);
+    default:
+      // .json and .js
+      return require(file);
```

Comment:
You should use `import` instead, so we can make this esm compatible.

Reply:
Ok sure, I just copied this over from the previous `getConfig()` approach. But an import also needs to be awaited so I've added it to https://github.com/renovatebot/renovate/pull/12649, could you please check that first @viceice?

Reply:
Thanks, I've merged main after the refactor PR now.